db.go

```diff
@@ -576,3 +576,58 @@ func (db *DB) loadIndexFromWAL() error {
 	}
 	return nil
 }
+
+// DeleteExpiredKeys scan the entire index in ascending order to delete expired keys.
+// It is a time-consuming operation, so we need to specify a timeout
+// to prevent the DB from being unavailable for a long time.
+func (db *DB) DeleteExpiredKeys(timeout time.Duration) {
+	db.mu.Lock()
+	defer db.mu.Unlock()
+
+	// set expiration time
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	now := time.Now().UnixNano()
+	go func(ctx context.Context) {
+		key := []byte{}
```

Comment:
We can save this key, and read directly from this key in the next call

Reply:
OK